src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java

```diff
@@ -207,6 +208,28 @@ public Visitable visit(MethodDeclaration method, Void p) {
     return result;
   }
 
+  @Override
+  public Visitable visit(Parameter para, Void p) {
+    if (insideTargetMethod) {
+      ResolvedType paraType = para.resolve().getType();
+      if (paraType.isReferenceType()) {
+        String paraTypeFullName =
+            paraType.asReferenceType().getTypeDeclaration().get().getQualifiedName();
+        usedClass.add(paraTypeFullName);
+        for (ResolvedType typeVariable : paraType.asReferenceType().typeParametersValues()) {
```

Comment:
Following the API, I'd name the loop variable `typeParameterValue` rather than `typeVariable`. (I also suggest changing the associated variables, like `typeVariableFullName` -> `typeParameterValueFullName`.)

Reply:
I'm also concerned that this code might be overspecialized to the case where classes that have type parameters are used as parameters. Can you add a test case where the target method also _defines_ the type variable in use, such as:
```
<T> void foo(List<T> list) { ... }
```

In particular, I'd like a test like this so that we can be certain that this code won't be triggered (creating a class named "T"!) in that case.

Reply:
No extraneous class is created, but Specimin does crash in this case, because it looks for the class `T` (i.e., a class with the type variable's name). I'll commit the test on this branch. (I don't think I'll have time to actually fix the bug right now, but maybe I'll do that in a bit.)